src/test/retrieve-profile.test.js

```diff
@@ -0,0 +1,77 @@
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import sinon from 'sinon';
+import Sinonchai from 'sinon-chai';
+import app from '../index';
+import JWTService from '../services/jwt.service';
+import UserController from '../controllers/user.controller';
+
+chai.use(Sinonchai);
+chai.use(chaiHttp);
+chai.should();
+
+const profileEndpoint = '/api/v1/user/profile';
+let userToken = `Bearer ${JWTService.generateToken({ id: 1, role: 'requester'})}`;
+
+describe('/RETRIEVE PROFILE', () => {
+
+    it('should allow only logged in users', (done) => {
+        chai.request(app)
+            .get(`${profileEndpoint}?rememberMe=yes`)
+            .set('Authorization', '')
+            .end((err, res) => {
+                res.should.have.status(401);
+                res.body.should.have.property('status').eql('error');
+                done();
+            })
+    });
+
+    it('should return 200 if profile is retrieved successfully', (done) => {
+        chai.request(app)
+            .get(`${profileEndpoint}?rememberMe=yes`)
+            .set('Authorization', userToken)
+            .end((err, res) => {
+                res.should.have.status(200);
+                res.body.should.have.property('status').eql('success');
+                done();
+            })
+    });
+
+    it('should return 500 if something goes wrong', (done) => {
```

Comment:
Don't you think something like
```
query: { rememberMe: 'yes' },
headers: { authorization: userToken' }
...

```
would have been more readable?

Reply:
Can you refer me to a particular line? Cant find this, hehehe

Reply:
`retrieve-profile.test.js` lines 41-53

Reply:
Hehe, seen it. Blame it on `Webstorm`, it forces you to put your properties on a different line from the braces (array & curly). And thus, it became a habit.